tests/src/test/scala/com/apollographql/scalajs/react/ReactApolloTest.scala

```diff
@@ -26,4 +30,34 @@ class ReactApolloTest extends AsyncFunSuite {
       )
     ))
   }
+
+  test("Can render data to string based on a query") {
+    js.Dynamic.global.fetch = UnfetchFetch
+
+    val gotDataPromise = Promise[Assertion]
+    val gotRenderedStringPromise = Promise[Assertion]
+    val link = new HttpLink(options = HttpLinkOptions(uri = "https://w5xlvm3vzz.lp.gql.zone/graphql"))
+    val cache = new InMemoryCache()
+    val client = new ApolloClient(options = js.Dynamic.literal(ssrMode = true, link = link, cache = cache))
+
+    ReactApolloServer.renderToStringWithData(
+      ApolloProvider(ApolloProvider.Props(client = client))(
+        Query(
+          CurrencyRatesQuery,
+          CurrencyRatesQuery.Variables("USD"),
+          queryOptions = ExtraQueryOptions(ssr = true)) { d =>
+          if (d.data.isDefined) {
+            gotDataPromise.success(assert(d.data.get.rates.exists(_.nonEmpty)))
+            println(JSON.stringify(d.data.get.rates.head))
+            div(JSON.stringify(d.data.get.rates.head))
+          } else ""
+        }
+      )
+    ).toFuture.foreach(html => {
+      gotRenderedStringPromise.success(assert(html.length > 0))
```

Comment:
Could this assertion be made more specific to test that the string contains the fetched data?